rest-service/manager_rest/resource_manager.py

```diff
@@ -1356,6 +1358,103 @@ def _finalize_create_deployment(self, deployment: models.Deployment):
         # RD-1602 will move this to the workflow:
         self._create_deployment_initial_dependencies(deployment)
 
+    @staticmethod
+    def get_deployment_parents_from_labels(labels):
+        parents = []
+        for label, value in labels:
+            if label == 'csys-obj-parent':
+                parents.append(value)
+        return parents
+
+    def verify_deployment_parent_labels(self, parents, deployment_id):
+        if not parents:
+            return
+        missing_parents = []
+        result = self.sm.list(
+            models.Deployment,
+            include=['id'],
+            filters={'id': lambda col: col.in_(parents)}
+        ).items
+        _existing_parents = [_parent[0] for _parent in result]
+        for parent in parents:
```

Comment:
This looks like an ideal candidate for using sets?
missing_parents = set(parents) - set(_existing_parents)

Reply:
(I think that's the right way to do it anyway- you have a test for this?)

Reply:
Yes there is a test case of missing parents